recipes/boost/all/conanfile.py

```diff
@@ -1160,11 +1160,14 @@ def add_libprefix(n):
                         self.cpp_info.components[module].requires.append("{0}::{0}".format(conan_requirement))
 
             if used_libraries != detected_libraries:
+                # TODO why do these sometime fail?
```

Comment:
These fail because the boost build script does not build a library if some dependency/requirement is not met.\

e.g. Boost.Locale is not built with `-o boost:without_locale=False -o boost:i18n_backend=icu -i icu:shared=False`.
This is clearly an error.

So these assertions catch errors.

Reply:
Also, currently boost is configured using `--without-xxx`.
So all modules are enabled except those explicitly removed.

In #3872 I have changed it to `--with-xxx`. Hopefully, this will only build those components explicitly enabled.

Reply:
But there seem to be false positive errors, I only need system, yet boost is not failing because of nowide. What can we do ASAP to get boost working?

Reply:
To get it working **NOW**, try building with `-o boost:extra_b2_flags=--without-nowide`.

Reply:
It's not a false positive btw, it detected a rogue library which should not have been there.